code/directors.py

```diff
@@ -0,0 +1,85 @@
+import sqlite3
+
+# Tengjast gagnagrunninum
+conn = sqlite3.connect('data/rotten_tomatoes.db')
+cursor = conn.cursor()
+
+# Keyra SQL-fyrirspurnir
+queries = [
+    """
+    CREATE TABLE IF NOT EXISTS top10_tomatometer_movies AS
+    SELECT 
+        movie_title, 
+        tomatometer_rating, 
+        audience_rating, 
+        directors, 
+        actors, 
+        original_release_date
+    FROM 
+        rotten_tomatoes_movies
+    WHERE 
+        actors LIKE '%Kate Winslet%' OR actors LIKE '%Leonardo DiCaprio%'
+    ORDER BY 
+        tomatometer_rating DESC
+    LIMIT 10;
+    """,
+    """
+    CREATE TABLE IF NOT EXISTS top10_audience_movies AS
+    SELECT 
+        movie_title, 
+        tomatometer_rating, 
+        audience_rating, 
+        directors, 
+        actors, 
+        original_release_date
+    FROM 
+        rotten_tomatoes_movies
+    WHERE 
+        actors LIKE '%Kate Winslet%' OR actors LIKE '%Leonardo DiCaprio%'
+    ORDER BY 
+        audience_rating DESC
+    LIMIT 10;
+    """,
+    """
+    CREATE TABLE IF NOT EXISTS all_movies_tomatometer_directors AS
+    SELECT 
+        movie_title, 
+        tomatometer_rating, 
+        audience_rating, 
+        directors, 
+        actors, 
+        original_release_date
+    FROM 
+        rotten_tomatoes_movies
+    WHERE 
+        directors IN (
+            SELECT DISTINCT directors 
+            FROM top10_tomatometer_movies
+        );
+    """,
+    """
+    CREATE TABLE IF NOT EXISTS all_movies_audience_directors AS
+    SELECT 
+        movie_title, 
+        tomatometer_rating, 
+        audience_rating, 
+        directors, 
+        actors, 
+        original_release_date
+    FROM 
+        rotten_tomatoes_movies
+    WHERE 
+        directors IN (
+            SELECT DISTINCT directors 
+            FROM top10_audience_movies
+        );
```

Comment:
```suggestion
SELECT 
    rtm.movie_title, 
    rtm.tomatometer_rating, 
    rtm.audience_rating, 
    rtm.directors, 
    rtm.actors, 
    rtm.original_release_date
FROM 
    rotten_tomatoes_movies rtm
JOIN 
    (SELECT directors FROM top10_audience_movies GROUP BY directors) tam
ON 
    rtm.directors = tam.directors;
```
Það er bæði eðlilegra og miklu hraðvirkara og skilvirkara að nota JOIN í stað þess að nota IN ásamt SELECT. Þetta dregur úr kostnaði við keyrslu, sérstaklega ef gagnagrunnurinn er stór, því IN með innföldu SELECT getur verið hægvirkt.